### PR TITLE
#1246 Add 'fill slide' in the right-click menu

### PR DESCRIPTION
**Pull request by junkiattan**
Added in the "fill-slide" functionality to the right-click context menu for existing images.

Tested on PPT2013 and PPT2016

Included a Functional Test for the function

Screenshot when existing image is selected
capture

Screenshot when non-image is selected
capture

### CODE REVIEW

**Comment by leeyh20** [comment]
Are there any tests for this? I have tested manually on PPT 2010.

File: PowerPointLabs/PowerPointLabs/ActionFramework/ShortcutsLab/FillSlide/FillSlideActionHandler.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using PowerPointLabs.ActionFramework.Common.Attribute;
+using PowerPointLabs.ActionFramework.Common.Extension;
+using PowerPointLabs.ActionFramework.Common.Interface;
+using PowerPointLabs.ShortcutsLab;
+using PowerPointLabs.TextCollection;
+
+namespace PowerPointLabs.ActionFramework.PasteLab
```
Comment (leeyh20) [inline PowerPointLabs/PowerPointLabs/ActionFramework/ShortcutsLab/FillSlide/FillSlideActionHandler.cs]: The namespace should be ShortcutsLab. Same for the rest.

File: PowerPointLabs/PowerPointLabs/ActionFramework/ShortcutsLab/FillSlide/FillSlideActionHandler.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        protected override void ExecuteAction(string ribbonId)
+        {
+            this.StartNewUndoEntry();
+
+            Microsoft.Office.Interop.PowerPoint.Selection currentSelection = this.GetCurrentSelection();
+            PowerPointLabs.Models.PowerPointSlide currentSlide = this.GetCurrentSlide();
+            float slideHeight = this.GetCurrentPresentation().SlideHeight;
+            float slideWidth = this.GetCurrentPresentation().SlideWidth;
+            // Send over to Fill operation to fill up the slide
```
Comment (leeyh20) [inline PowerPointLabs/PowerPointLabs/ActionFramework/ShortcutsLab/FillSlide/FillSlideActionHandler.cs]: The comment is a little unnecessary to me.

File: PowerPointLabs/PowerPointLabs/ActionFramework/ShortcutsLab/FillSlide/FillSlideEnabledHandler.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using PowerPointLabs.ActionFramework.Common.Attribute;
+using PowerPointLabs.ActionFramework.Common.Extension;
+using PowerPointLabs.ActionFramework.Common.Interface;
+using PowerPointLabs.TextCollection;
+using PowerPointLabs.Utils;
+
+namespace PowerPointLabs.ActionFramework.PasteLab
```
Comment (leeyh20) [inline PowerPointLabs/PowerPointLabs/ActionFramework/ShortcutsLab/FillSlide/FillSlideEnabledHandler.cs]: Namespace should be ShortcutsLab.

File: PowerPointLabs/PowerPointLabs/ActionFramework/ShortcutsLab/FillSlide/FillSlideEnabledHandler.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+{
+    [ExportEnabledRibbonId(ShortcutsLabText.FillSlideTag)]
+    class FillSlideEnabledHandler : EnabledHandler
+    {
+        protected override bool GetEnabled(string ribbonId)
+        {
+            //Gets the current selection
+            Microsoft.Office.Interop.PowerPoint.Selection currentSelection = this.GetCurrentSelection();
+            //Checks if everything currently selected are either shapes or pictures and if number of objects selected equals one
```
Comment (leeyh20) [inline PowerPointLabs/PowerPointLabs/ActionFramework/ShortcutsLab/FillSlide/FillSlideEnabledHandler.cs]: Comments seem unnecessary to me, considering it's quite readable from the method names,

File: PowerPointLabs/PowerPointLabs/ShortcutsLab/FillSlide.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            if (ppShapeToFitSlide.AbsoluteWidth < slideWidth)
+            {
+                ppShapeToFitSlide.AbsoluteWidth = slideWidth;
+            }
+            ppShapeToFitSlide.VisualCenter = new System.Drawing.PointF(slideWidth / 2, slideHeight / 2);
+
+            CropLab.CropToSlide.Crop(shapeToFitSlide, slide, slideWidth, slideHeight);
+        }
+
+        private static PowerPoint.Shape GetShapeFromSelection(PowerPoint.Selection selection)
```
Comment (leeyh20) [inline PowerPointLabs/PowerPointLabs/ShortcutsLab/FillSlide.cs]: Why do you need to create two overloading methods for this if all the methods are only called once? Could be combined to one method?

**Comment by junkiattan** [comment]
@leeyh20 I have addressed all of your comments;)

**Comment by initialshl** [comment]
Can you attach some screenshots in this PR? Thanks

**Comment by junkiattan** [comment]
@initialshl I have added the screen shots! Awaiting your approval;)

**Pull request merged**